### PR TITLE
gui things, help me get values from the lists and suggest other things to do!

### PR DESCRIPTION
**Pull request by ionisea**
we may want to use values for each shape instead;
0 = line
1= circle
etc

either way please make sure the other things are to your liking and mr seibel please make sure this won't break anything else, I didn't know where to put it besides index

### CODE REVIEW

**Comment by lashlockbhs** [comment]
"document.getElementById("").value" to get value of input?

File: projects/physics/public/style.css [hunk 1]
```diff
@@ -9,8 +9,9 @@ body {
   width: 100%;
   text-align: center;
 }
-
 canvas {
-  margin: 1in auto;
+  margin: 1in bottom;
```
Comment (gigamonkey) [inline projects/physics/public/style.css:13]: eh? what does that do?

File: projects/physics/public/index.html [hunk 2]
```diff
@@ -7,6 +7,37 @@
   <link href="style.css" rel="stylesheet" type="text/css">
   <script src="script.js" type="module"></script>
 </head>
+<labels>
+  <label for="density"> Density (constant), kg/pixel:</label>
+  <input type="number" id="density" name="density"
```
Comment (gigamonkey) [inline projects/physics/public/index.html:12]: You may also want to look at `<input type="range">`

**Review by gigamonkey — COMMENTED** [review]
I'm pretty sure `bottom` is not a valid value for the `margin` property. In which case it probably gets turned into 0 which may look fine. But I'm not sure what you're actually trying to do there. Everything else looks reasonable.

File: projects/physics/public/style.css (hunk 1 above)
Comment (ionisea) [reply projects/physics/public/style.css:13]: nothing, removed

File: projects/physics/public/index.html (hunk 2 above)
Comment (ionisea) [reply projects/physics/public/index.html:12]: I tried this but it doesn't really fit the idea of this we've had and I think im going to keep using number

**Comment by ionisea** [comment]
@lashlockbhs i need you to figure out the on key down stuff I have no clue how it works

File: projects/physics/public/perfectShapes.js [hunk 3]
```diff
@@ -0,0 +1 @@
+document.getContext('')
```
Comment (lashlockbhs) [inline projects/physics/public/perfectShapes.js:1]: remove cause wrong

File: projects/physics/public/script.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ import {
   height, 
   animate, 
   now, 
+  onKeyDown,
```
Comment (lashlockbhs) [inline projects/physics/public/script.js:9]: change name. to similar to other func

File: projects/physics/public/script.js [hunk 5]
```diff
@@ -170,14 +171,17 @@ registerOnclick((x, y) => {
 
 registerOnKeyDown(() => {
   if (!animateStart) {
-    objArray.push(new Shape(10, [vector(0, 0)], vertices))
+    const area = sigma(0, objArray.length - 2, (i) => objArray[i].x * objArray[i + 1].y - objArray[i + 1].x * objArray[i].y) / 2;
+    objArray.push(new Shape(10, [vector(0, 0)], vertices, area * document.getElementById('density').getAttribute('value')));
```
Comment (lashlockbhs) [inline projects/physics/public/script.js:175]: move function to math.js.

**Review by lashlockbhs — CHANGES_REQUESTED** [review]
Fix what I suggested and figure out how do register functions based on the specific key pressed.

**Review by szetingchen — COMMENTED** [review]
lgtm



(i didnt look, if it breaks its lukes fault)

File: projects/physics/public/script.js (hunk 5 above)
Comment (ionisea) [reply projects/physics/public/script.js:174]: done

File: projects/physics/public/script.js (hunk 4 above)
Comment (ionisea) [reply projects/physics/public/script.js]: removed from import, I don't know how it works

File: projects/physics/public/index.html (hunk 2 above)
Comment (gigamonkey) [reply projects/physics/public/index.html]: 👍🏻

File: projects/physics/public/script.js [hunk 6]
```diff
@@ -169,15 +169,20 @@ registerOnclick((x, y) => {
...
-    objArray.push(new Shape(10, [vector(0, 0)], vertices))
+  if (!animateStart && objArray.length < 5) {
+    const area = shapeArea(vertices)
+    objArray.push(new Shape(10, [vector(0, 0)], vertices, area * document.getElementById('density').getAttribute('value')));
     objArray[objArray.length - 1].drawShape()
     drawFilledCircle(objArray[objArray.length - 1].center.x, objArray[objArray.length - 1].center.y, 2.5, "red")
     vertices = []
-    animateStart = objArray.length >= 3 ? true : false
+  } else {
+    animateStart = animateStart ? false : true
```
Comment (gigamonkey) [inline projects/physics/public/script.js:179]: You can also write this more concisely and more idiomatically as:

```javascript
    animateStart = !animateStart; 
```

File: projects/physics/public/index.html [hunk 7]
```diff
@@ -7,7 +7,38 @@
...
+  <label for="density"> Density (constant), kg/pixel:</label>
+  <input type="number" id="density" name="density" value="100"
+       min="0">
+       <!-- stolen from mdn xd trolled diots https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number-->
+       
+  <label for="radius"> -- Radius for perfect shapes, constant --</label>
+  <input type="number" id="radius" name="radius"
+       min="0" max="500">
+            <!-- ALSO stolen from mdn https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number-->
+            <label for="perfectShape">-- Perfect Shapes (check README.md) --</label>
```
Comment (gigamonkey) [inline projects/physics/public/index.html:20]: The `for` attribute of a `label` is supposed to be the value of the `id` attribute of the element it's a label for. I think you mean this to be `shape-select`?

Comment (gigamonkey) [inline projects/physics/public/index.html:20]: This indentation is a bit wonky. I think the editor's "Format document" command would clean it up.

File: projects/physics/public/script.js [hunk 8]
```diff
@@ -32,7 +33,7 @@ import {
 
 const canvas = document.getElementById('screen');
 setCanvas(canvas);
-const FPS = 12 // frames per second
+const FPS = 12 // frames per second, consider changing to ms/frame
```
Comment (gigamonkey) [inline projects/physics/public/script.js:36]: You don't seem to actually be using this for anything. You should either delete it or finish whatever idea you had.

**Review by gigamonkey — COMMENTED** [review]
All looks pretty reasonable to me.

**Review by lashlockbhs — APPROVED** [review]
lgtm.

**Comment by ionisea** [comment]
> "document.getElementById("").value" to get value of input?

ill figure this out later

**Comment by ionisea** [comment]
>  figure out how do register functions based on the specific key pressed.
thats your job

**Review by szetingchen — APPROVED** [review]
lgtm

**Pull request merged**